pre-commit

```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+# Run SQLFluff on staged SQL files
+git diff --name-only --cached | grep '\.sql$' | xargs sqlfluff lint
```

Comment:
I would suggest we use the pre-commit framework instead of rolling our own script here. We should be able to just follow [this guide](https://pre-commit.com/index.html#quick-start) to create a `.pre-commit.yaml` file that looks something like this:

```md
repos:
-   repo: https://github.com/sqlfluff/sqlfluff
    rev: 3.0.3
    hooks:
    -   id: sqlfluff-fix
```

The only thing I'm not sure about is whether this approach will pick up the `.sqlfluff` config, but I think it should.

The advantages of this approach are that the pre-commit framework will handle management of sqlfluff for you, and it can be [easily configured](https://pre-commit.com/index.html#3-install-the-git-hook-scripts) to automatically run on every local commit. Also, we can easily add a GitHub Action to run pre-commit and require a passing pre-commit run for any PRs.

Reply:
This looks like a great direction.  I will give it a shot.

Reply:
@acontry Updated the pre-commit method and it worked locally.  I'm going to merge and try to set-up the GitHub Action.